src/servers/FtpServer.cc

```diff
@@ -376,7 +376,7 @@ Ftp::Server::listenForDataConnection()
     conn->local = transparent() ? port->s : clientConnection->local;
     conn->local.port(0);
     const char *const note = uri.c_str();
-    comm_open_listener(SOCK_STREAM, IPPROTO_TCP, conn, note);
+    OpenBoundedListener(SOCK_STREAM, IPPROTO_TCP, conn, note);
```

Comment:
Hm... Is there any use case where a comm_open_listener() caller would want to use `IP_BIND_ADDRESS_NO_PORT`? I suspect there are no such use cases in the current code and there is no reason to believe that such cases will appear in the future: If I am opening a _listening_ port, then I do not _need_ (i.e. cannot benefit from) `IP_BIND_ADDRESS_NO_PORT`, right? If that is the case, then we do not need OpenBoundedListener() AFAICT -- its functionality can be moved into comm_open_listener() itself. If you agree, please adjust the code accordingly.

P.S. I should have realized the mutually exclusive nature of the new feature and comm_open_listener() in the original review. Sorry for this "extra" iteration.

Reply:
Ah .. I hadn't realized that the comm_open_listener() is only called for listen sockets which will never need the IP_BIND_ADDRESS_NO_PORT sockopt. Removed the extra wrapper function.